### PR TITLE
[JUJU-2567] Refactor secret apis to support upcoming cmr support

### PR DESCRIPTION
**Pull request by wallyworld**
To prepare for cmr secrets support, some refactoring work is needed. The key changes are to add a model-uuid to the secret URL and to move code from the secretsmanager facade to apiserver/common.

Secret URLs now support `secret://<model-uuid>/<id>`. The model locator will be used to allow the controller to direct secret requests to the relevant target model (which may be on a different controller).

The facade calls have been changed to take a backend id as a parameter to fetch backend config, instead of fetching all config and resolving caller side. APIs to get secret content now also return the config for the relevant backend (if needed) as part of the api call, instead of needing to separately fetch the backend config.
For the case where the authenticated secret owner is accessing its own secret metadata, we remove the explicit "can manage" permission check to allow for the necessary grouping of common logic.

## QA steps

Deploy `juju-qa-dummy-source` and `juju-qa-dummy-sink` and relate.
Add a vault back end
Create a secret for dummy-source and grant access to dummy-sink
Ensure dummy-sink can see the secret
Revoke access and check permission denied
Add a second dummy-source unit and ensure it can access the secret but not change it

### CODE REVIEW

**Review by ycliuhw — APPROVED** [review]
LGTM, tyvm!

File: apiserver/facades/schema.json [hunk 1]
```diff
@@ -39954,6 +39958,26 @@
                         "model-controller",
                         "model-uuid",
                         "model-name",
+                        "draining"
+                    ]
+                },
+                "SecretBackendConfigResults": {
```
[the remark this answers is not in the document]
Comment (wallyworld) [reply apiserver/facades/schema.json:39964]: No because this is for 3.2 and we will be auto upgrading migrated models

**Pull request merged**